### ISSUE TITLE
GraphAPI path: ONetworkProtocolBinary: Cannot serialize record

### ISSUE THREAD

**Issue by jmau2002** [orientechnologies/orientdb#7308]
## Expected behavior  
Path returned to client

## Actual behavior  
Error on server side in ONetworkProtocolBinary: "Cannot serialize record"
2017-04-05 13:36:56:157 WARNI {db=GratefulDeadConcerts} Cannot serialize record: [v(Entity)[#81:0], v(Entity)[#83:0], v(Entity)[#81:1]] [ONetworkProtocolBinary]

## Steps to reproduce  
Setup: 
- OrientDB as server, GratefulDeadConcerts database
- java app as  client, accessing the remote database using tinkerpop 2.6 gremlin-groovy

```
// Gremlin expression which returns a path
OrientGraph graph = new OrientGraph("remote:localhost/GratefulDeadConcerts");
String gremlinCmd2 = "g.v('81:0').both.loop(1){it.loops<5}{it.object.name=='Robert'}.simplePath.path";
Iterable<?> allVertices =  graph.command(new OCommandGremlin(gremlinCmd2)).execute();
Iterator<?> iterator = allVertices.iterator();
while (iterator.hasNext()) {
        Object v = iterator.next();    // error here
       System.out.println(v);
}
```
On the client side, I receive in iterator.next() 
java.util.NoSuchElementException
	at com.tinkerpop.blueprints.impls.orient.OrientDynaElementIterator.next(OrientDynaElementIterator.java:57)

On the server side I see 
2017-04-05 13:39:04:287 WARNI {db=GratefulDeadConcerts} Cannot serialize record: [v(Entity)[#81:0], v(Entity)[#82:0], v(Entity)[#81:1]] [ONe
tworkProtocolBinary]

Why can't the path be serialised? 

I saw [1] which was solved in 2.2 (10/2016), so it's probably not the same problem. 

[1] https://github.com/orientechnologies/orientdb/issues/5579

**Comment by Zakjholt** [comment]
Also experiencing the same error using any gremlin traversal query with the .groupBy and .cap pipes. 
OrientDB version 2.2.20

**Comment by tglman** [comment]
hi @jmau2002, @Zakjholt 

The reason why you get this error is because we do not support result sets of collection types in the binary protocol, I just pushed a fix in the integration between orient and gremlin that convert this kind of results in result set of documents, that are correctly handled by the binary protocol fixing the problem.

the fix will be release in 2.2.21

Regards

**Comment by tglman** [comment]
Hi,

The 2.2.21 has been released last week, closing this.

Re-open if the problem happen again.

Regards
